test_common/Test_Common_ArithTraits.hpp

```diff
@@ -395,6 +395,11 @@ class ArithTraitsTesterBase {
         success = 0;
        }
     }
+    if ( ! std::is_same< ScalarType, typename std::result_of<decltype(&AT::infinity)()>::type >::value )
```

Comment:
Could use a static assertion instead if you like.

Reply:
Hm, do we not test the case where `ScalarType` is `std::complex<R>`?  I guess not :-)

Why don't we just write `decltype(AT::infinity())` instead of the `std::result_of` expression?

Reply:
No good reason.  Will change.

Reply:
btw I'm happy with the PR -- I don't mean to make you do more work -- but I'll be happy to see further improvements and will gladly review them :-)

Reply:
Any reason for delaying the merge?